### ISSUE TITLE
Does not work in strict mode!

### ISSUE THREAD

**Issue by tankerkiller125** [felixge/node-require-all#26]
Using strict mode (say for using ECMASCRIPT6 classes causes an error. 

```
SyntaxError: Unexpected token =
    at exports.runInThisContext (vm.js:53:16)
    at Module._compile (module.js:414:25)
    at Object.Module._extensions..js (module.js:442:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:313:12)
    at Function._load (/home/tankerkiller125/clive-servers/websockets/node_modules/pmx/lib/transaction.js:62:21)
    at Module.require (module.js:366:17)
    at require (module.js:385:17)
    at /home/tankerkiller125/clive-servers/websockets/node_modules/require-all/index.js:41:50
    at Array.forEach (native)
    at requireAll (/home/tankerkiller125/clive-servers/websockets/node_modules/require-all/index.js:23:9)
    at Object.<anonymous> (/home/tankerkiller125/clive-servers/websockets/endpoints/chat/index.js:5:35)
    at Module._compile (module.js:435:26)
    at Object.Module._extensions..js (module.js:442:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:313:12)
```

**Comment by dougwilson** [comment]
Hi @tankerkiller125, please feel free to make a pull request to fix the issue! Otherwise, I need complete details on how to reproduce the issue, including the version of this module you are using, the version of node.js, complete code that reproduces the issue, and instructions on how to reproduce.

Thanks for the report!

**Comment by dougwilson** [comment]
Hi @tankerkiller125, I'm not able to reproduce, and I haven't heard back on my question in 24 days. I'm going to close this issue, but if you can provide a pull request to fix the issue or provide complete details on how to reproduce the issue, including the version of this module you are using, the version of node.js, complete code that reproduces the issue, and instructions on how to reproduce, I can always re-open!

**Comment by mcodev1** [comment]
@dougwilson I'm also encountering this error. Specifically, my case has to do with ES6 function default parameters. All I've done to isolate the issue is edit this file: https://github.com/felixge/node-require-all/blob/master/test/controllers/main-Controller.js in the `node-require-all` test case to the following:

```js
exports.index = 1;
exports.show  = 2;
exports.add   = 3;
exports.edit  = 4;
exports.fn    = function(param = "default") { return param; };
```

Running `npm test` then generates the following output:

```
mstahl:~/code/node-require-all (master)> npm test

> require-all@2.0.0 test /Users/mstahl/code/node-require-all
> node test/test.js

/Users/mstahl/code/node-require-all/test/controllers/main-Controller.js:6
exports.fn    = function(param = "default") { return param; };
                               ^

SyntaxError: Unexpected token =
    at exports.runInThisContext (vm.js:53:16)
    at Module._compile (module.js:413:25)
    at Object.Module._extensions..js (module.js:452:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)
    at /Users/mstahl/code/node-require-all/index.js:41:50
    at Array.forEach (native)
    at requireAll (/Users/mstahl/code/node-require-all/index.js:23:9)
npm ERR! Test failed.  See above for more details.
```

**Comment by dougwilson** [comment]
Hi @mcodev1, what happens if you run the following command?
```
node -e 'require("./test/controllers/main-Controller.js")'
```

**Comment by mcodev1** [comment]
Same error.

```
mstahl:~/code/node-require-all (master)> node -e 'require("./test/controllers/main-Controller.js")'
/Users/mstahl/code/node-require-all/test/controllers/main-Controller.js:5
exports.fn    = function(param = "default") { return param; };
                               ^

SyntaxError: Unexpected token =
    at exports.runInThisContext (vm.js:53:16)
    at Module._compile (module.js:413:25)
    at Object.Module._extensions..js (module.js:452:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)
    at [eval]:1:1
    at Object.exports.runInThisContext (vm.js:54:17)
    at Object.<anonymous> ([eval]-wrapper:6:22)
```

I'm using Node v4.1.1

**Comment by dougwilson** [comment]
Hi @mcodev1, then that is expected. All this module does is `require()` each file. If Node.js cannot load the file, then I'm not sure what you would expect this module to do?

**Comment by mcodev1** [comment]
Valid point :)

I hadn't checked the normal `require()` behavior. Thanks for the quick response.

It seems this wasn't implemented in Node until v6.0.0
